dss/events/handlers/index.py

```diff
@@ -218,9 +221,21 @@ def notify(subscription_id: str, subscription: dict, bundle_id: str, logger):
             "bundle_version": bundle_version
         }
     }
-    # TODO (mbaumann) Ensure webhooks are only delivered over verified HTTPS (unless maybe when running a test)
     callback_url = subscription['callback_url']
-    response = requests.post(callback_url, json=payload)
+
+    assert urlparse(callback_url).scheme in {"http", "https"}
+    if os.environ["DSS_DEPLOYMENT_STAGE"] not in {"dev", "staging"}:
+        hostname = urlparse(callback_url).hostname
+        for family, socktype, proto, canonname, sockaddr in socket.getaddrinfo(hostname, port=None):
+            assert ipaddress.ip_address(sockaddr[0]).is_global  # type: ignore
+        assert urlparse(callback_url).scheme == "https"
```

Comment:
The conditions asserted here can/should also be checked at the time the subscription is registered, and the registration should be rejected if these conditions are not me. It is preferable to not allow an invalid subscription into the system than to allow a subscription for which the notifications will always fail.
@Bento007/I can help with these changes if you agree with the above.

Reply:
Actually, they should be checked in both places.

Reply:
Yes, that is what I intended to communicate. Thanks!

Reply:
Filed #550 to deal with this later.